rewrite-java-test/src/test/java/org/openrewrite/java/cleanup/RemoveCallsToObjectFinalizeTest.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2022 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.cleanup;
+
+import org.junit.jupiter.api.Test;
+import org.openrewrite.test.RecipeSpec;
+import org.openrewrite.test.RewriteTest;
+
+import static org.openrewrite.java.Assertions.java;
+
+class RemoveCallsToObjectFinalizeTest implements RewriteTest {
+
+    @Override
+    public void defaults(RecipeSpec spec) {
+        spec.recipe(new RemoveCallsToObjectFinalize());
+    }
+
+    @Test
+    void removeSimpleFinalize() {
+        rewriteRun(
+          java(
+            """
+              public class A {
+              
+                      @Override
+                      protected void finalize() {
+                          super.finalize();
+                      }
```

Comment:
I need some help here. 

While the call to super.finalize is removed, the call to 

```java

doAfterVisit(new EmptyBlock());
```

Is not removing the method invocation

Reply:
Had to think about this one for a bit; it's good that you've captured it in a unit test!

Looking at [EmptyBlockVisitor](https://github.com/openrewrite/rewrite/blob/v7.35.0/rewrite-java/src/main/java/org/openrewrite/java/cleanup/EmptyBlockVisitor.java) & [EmptyBlockTest](https://github.com/openrewrite/rewrite/blob/v7.35.0/rewrite-java-test/src/test/java/org/openrewrite/java/cleanup/EmptyBlockTest.java) it appears that one does not remove methods with empty overrides. And thinking about the various ways in which people can write their code, I'm not sure it should.  Consider a case where `B` extends `A`, and `B` overrides a method in `A` with an empty body, without a call to `super`. That might be intentional if `B` wants to skip the logic in `A`s super method.

In the test case you've created here it would make sense to remove the overridden method, as there's only a call to `super` in there, but I don't think we'll find such constructs often in the wild. What I would propose then is to (at least for now) never remove calls to `super.finalize()`, as those could very well be intentional. 

Does that make sense to you?

Reply:
Thank you. I updated it